python/lsst/sims/maf/slicers/baseSpatialSlicer.py

```diff
@@ -59,7 +60,10 @@ def __init__(self, verbose=True, lonCol='fieldRA', latCol='fieldDec',
         self.slicePoints['ra'] = None
         self.slicePoints['dec'] = None
         self.nslice = None
+        self.shape = None
         self.plotFuncs = [BaseHistogram, BaseSkyMap]
+        # Values in self.slicePoints to pass whole to the metrics
+        self.bulkSlicePointKeys = []
```

Comment:
So I started fiddling, and eventually went whole-hog and moved the bin information over to the metrics, and gave the metrics a shape attribute as well.  This way, you can even run regular metrics and vector metrics with the same slicer!